sympy/utilities/lambdify.py

```diff
@@ -174,6 +176,12 @@ def lambdify(args, expr, modules=None, printer=None, use_imps=True,
        This function uses ``exec``, and thus shouldn't be used on
        unsanitized input.
 
+    .. warning::
+       This function uses ``enumerate`` on ``args``, thus you should
```

Comment:
Rather than having a warning note here we could use the versionadded directive to say that passing a set is deprecated similar to this:
https://github.com/sympy/sympy/blob/87cca59f3507e1052631da0819edd0cfa1baadc0/sympy/printing/latex.py#L2925-L2926

Reply:
The messaging here is inconsistent. The version changed part says that `set` is no longer supported and then the warning part warns against the use of `set`.

I think that this should be a single versionchanged directive that says `"Passing a set for the *args* parameter is deprecated as sets are unordered. Use an ordered iterable such as a list or tuple."`